### DISCUSSION TITLE
Detecting changes to autograding scripts

### DISCUSSION THREAD

**Discussion by pinventado** [community/community#17777] [GitHub Education]
I'm following up on a conversation from the GitHub education forum (https://education.github.community/t/prevent-students-from-changing-autograding-tests/53718/16)

Are there any updates on rolling out the feature described by @d12 that checks if the GitHub actions files were modified?

**Comment by JBraunsmaJr** [comment]
I had the idea of using checksums against the files but then realized the `autograding.json` exists inside the repository, showing all the things. 

Unfortunately, I tested it and found I can modify it! So nothing stops students from tinkering with that approach either. Right now I can't think of an automated way of detecting changes.... 

The workaround for now can be to check the box for the feedback PR. If a student modifies the file you can easily see it in the files changed tab.

Kinda feels like an oversight that they don't have tampering measures out of the box.

**Comment by diamondburned** [comment]
This is not an answer, but I'm posting it as a proof-of-concept. You can verify the GPG signature of the latest commit made to `.github` to ensure that it's committed by a trusted user (like the GitHub Classroom bot).

I made a proof-of-concept for this at [diamondburned/autograde-guard](https://github.com/diamondburned/autograde-guard/), but it's far from ideal. So far, it's able to detect tampers to `.github` though. I currently have rollback implemented, but I haven't thoroughly tested it yet.

**Comment by agilebotanist** [comment]
Hello, I'm new to classrooms, but here are some ideas on preventing the effects of changing autograding workflows.

So far, the only possible solution that we have found is to download all repositories after the deadline and overwrite the settings files in post-processing. Are there any examples that we can use as inspiration?

--- context ---
We are creating labs for 300+ students for an advanced course on software quality and CI with GitHub Actions.

We chose to try GitHub Classroom with Authograding in Python. 

We immediately noticed that Autograding action could be easily diverted by students who know a bit about GitHub Actions. If one changes a few settings, all assignments pass by magic. It is also possible to add extra points. 

I'm looking for methods to protect auto-grading settings and obfuscate certain tests so that students cannot guess solutions by just copying our tests.

What are the best practices?
